### PR TITLE
[PLATFORM-946] Restore sidebar version

### PR DESCRIPTION
**Pull request by juhah**
Testing in PR build

### CODE REVIEW

File: app/src/shared/utils/diagnostics.js [hunk 1]
```diff
@@ -10,8 +10,8 @@ global.streamr = Object.assign(global.streamr || {}, {
         return {
             userAgent: navigator.userAgent,
             environment: process.env.NODE_ENV,
-            version: process.env.GIT_VERSION,
-            branch: process.env.GIT_BRANCH,
+            version: process.env.GIT_VERSION || process.env.TRAVIS_TAG,
+            branch: process.env.GIT_BRANCH || process.env.TRAVIS_BRANCH,
```
Comment (timoxley) [inline app/src/shared/utils/diagnostics.js]: 👍

**Comment by timoxley** [comment]
I guess testing this is blocked until the CI issue is fixed

**Comment by juhah** [comment]
Possibly works 🤷‍♂ testable at https://d3oakyxnh2wtk4.cloudfront.net/

PR env is flaky but I was able to login and see this in the sidebar:
Screen Shot 2019-07-23 at 13 19 55

**Comment by timoxley** [comment]
@juhah hrm, `HEAD` as the branch and some weird SHA `974b1b0` doesn't seem very useful.

image

**Comment by timoxley** [comment]
@juhah maybe it's actually picking up `GIT_` vars, try prioritising the `TRAVIS_` env vars over the `GIT_` ones if they exist (basically reverse the `||`)

File: app/src/editor/shared/components/Sidebar/Header.jsx [hunk 2]
```diff
@@ -12,22 +12,38 @@ export type Props = {
...
-        <div className={styles.appInfo}>
-            Streamr App v2.0.5 Beta
+const Header = ({ title, onClose }: Props) => {
+    const { version, branch } = global.streamr.info()
+    const isMaster = branch === 'master'
+    // version e.g. v2.0.5-926-g7e20dd2eb
+    const [versionNumber, , hash] = version.split('-')
+
+    // hash minus leading 'g', not shown on master
+    const displayHash = isMaster ? '' : hash.slice(1)
```
Comment (fonty1) [inline app/src/editor/shared/components/Sidebar/Header.jsx]: Would it make sense to move this logic creating `displayHash, displayBranch` to `app/src/shared/utils/diagnostics.js` and export it there?

Comment (timoxley) [reply app/src/editor/shared/components/Sidebar/Header.jsx]: Not a bad idea

**Comment by timoxley** [comment]
Hrm, no change. 


image

Maybe it needs `TRAVIS_PULL_REQUEST_BRANCH` for prs and `process.env.TRAVIS_BRANCH` for the other ones?

https://github.com/streamr-dev/streamr-platform/blob/3912a85df09d090d488776e0c3d42d6a73724f3a/.travis.yml#L52-L73

https://docs.travis-ci.com/user/environment-variables/

And maybe `TRAVIS_COMMIT` or `TRAVIS_PULL_REQUEST_SHA` if `TRAVIS_TAG` is `HEAD`? Blegh.

**Comment by juhah** [comment]
@timoxley I add them to the header, let's see what we get.

**Comment by juhah** [comment]
@timoxley It doesn't give anything actually:
Screen Shot 2019-07-25 at 10 27 43

**Comment by timoxley** [comment]
@juhah need to add them to the webpack Env plugin config

**Comment by juhah** [comment]
This looks better:
Screen Shot 2019-07-26 at 11 10 11

**Comment by juhah** [comment]
https://d1pdjq1d4bk0z6.cloudfront.net/

Screen Shot 2019-07-31 at 13 36 40

@fonty1 @timoxley Moved the info logic to `diagnostics.js`, added branching based on Travis builds.

**Comment by juhah** [comment]
@fonty1 I shortened the commit string to 8 chars. Let's check in PR build and then I'll merge and see the staging build. I think we are already back to using tagged builds, at least I released the latest version in June.

**Pull request merged**